blocks/bricks/__init__.py

```diff
@@ -553,8 +554,14 @@ def __init__(self, activations, dims, **kwargs):
         self.linear_transformations = [Linear(name='linear_{}'.format(i))
                                        for i in range(len(activations))]
         # Interleave the transformations and activations
-        application_methods = [brick.apply for brick in interleave(
-            [self.linear_transformations, activations]) if brick is not None]
+        application_methods = []
+        for entity in interleave([self.linear_transformations, activations]):
+            if entity is None:
+                continue
+            if isinstance(entity, Brick):
+                application_methods += [entity.apply]
```

Comment:
Please use `.append`, which should be much faster; I think list concatenation causes a new list to be created instead of the old one being appended to.